src/txn/txn_log.c

```diff
@@ -594,28 +595,40 @@ __wt_txn_truncate_log(
          * If the user provided a start cursor key (i.e. local_start is false) then use the original
          * key provided.
          */
-        if (!local_start && orig_start_key != NULL) {
+        if (F_ISSET(trunc_info, WT_TRUNC_EXPLICIT_START)) {
+            WT_ASSERT_ALWAYS(session, trunc_info->orig_start_key != NULL,
+              "Logging truncate operation with explicit start to range and empty key");
```

Comment:
I think this assertion message is confusing, I would prefer something like "Attempted to log truncate operation with explicit start to range but empty key".

Reply:
I agree it's confusing. My suggestion would be something like "Truncate log operation with explicit range start has empty original key."

If you choose to change this message, all the others should be similarly changed.